docs/design/architecture.md

```diff
@@ -2,128 +2,353 @@
+1. The [agent](#agent) is launched inside the VM.
+1. The agent creates a container environment to host the [workload](#workload).
+1. The agent spawns the workload inside the container environment.
+1. Control is returned to the user running `ctr`.
+
+   If the user requested it, the standard I/O streams of the workload
+   running inside the container in the VM are connected to the user's
+   terminal.
+
+Further details of these steps are provided in the sections below.
+
+### Container shutdown
 
-How Kata Containers maps container concepts to virtual machine technologies, and how this is realized in the multiple
-hypervisors and VMMs that Kata supports is described within the [virtualization documentation](./virtualization.md)
+There are two possible ways for the container environment to be
+destroyed:
+
+- When the [workload](#workload) exits.
+- When the container manager forces the container to be deleted.
+
+#### Workload exit
+
+The [agent](#agent) will detect when the [workload](#workload) process
+exits, capture its exit code and return it to the runtime, which
+passes it back to the container manager. Once the workload has fully
+exited, the VM is no longer needed and the runtime cleans up the
+environment (which includes killing the [hypervisor](#hypervisor)
+process).
+
+> **Note:**
+>
+> When [agent tracing is enabled](../tracing.md#agent-shutdown-behaviour),
+> the shutdown behaviour is different.
+
+#### Container manager requested shutdown
+
+If the container manager requests the container be deleted, the
+[runtime](#runtime) will signal the agent by sending it a
+[`DestroySandbox` ttRPC API](../../src/agent/protocols/protos/agent.proto) request.
 
 ## Guest assets
 
-The hypervisor will launch a virtual machine which includes a minimal guest kernel
-and a guest image.
+To create a VM to host a container, Kata Containers launches a
+hypervisor. The hypervisor needs two assets to create a VM: a kernel
+and an image file.
 
 ### Guest kernel
 
-The guest kernel is passed to the hypervisor and used to boot the virtual
-machine. The default kernel provided in Kata Containers is highly optimized for
-kernel boot time and minimal memory footprint, providing only those services
-required by a container workload. This is based on a very current upstream Linux
-kernel.
+The Linux [guest kernel](../../tools/packaging/kernel)
+is passed to the hypervisor and used to boot the VM.
+The default kernel provided in Kata Containers is highly optimized for
+kernel boot time and minimal memory footprint, providing only those
+services required by a container workload. This is based on a very
+current upstream Linux kernel.
 
 ### Guest image
 
-Kata Containers supports both an `initrd` and `rootfs` based minimal guest image.
+The hypervisor uses an image file which provides a minimal root
+filesystem used by the guest kernel to boot the VM and host the Kata Container.
+
+Kata Containers supports both initrd and rootfs based minimal guest images.
```

Comment:
I'd make a huge note there saying "depending on the hypervisor used", as I don't think we can use initrd with all the VMMs, neither I think we maintain tests with that (although Red Hat uses initrd for its product).